bin/s2nd.c

```diff
@@ -556,6 +573,14 @@ int main(int argc, char *const *argv)
         case 'L':
             key_log_path = optarg;
             break;
+        case 'P':
+            if (s2n_init_set == 0) {
+                GUARD_EXIT(s2n_init(), "Error running s2n_init()\n");
+                s2n_init_set = 1;
+            }
+            GUARD_EXIT(s2n_setup_external_psk(conn_settings.psk_list, &conn_settings.psk_idx, optarg),
+                       "Error setting external PSK parameters\n");
+            break;
```

Comment:
What happens if I run "s2nd -P blah,blah,blah -f"? You're introducing an ordering requirement on the arguments.

I think you would probably be better off saving the optargs in an array instead and parsing them later.